File: src/my_cpp_pkg/src/add_two_ints_serv.cpp
#include "rclcpp/rclcpp.hpp" //importing the ros2 lib if it shows a error open vs from the terminal
#include "example_interfaces/srv/add_two_ints.hpp"
using namespace std::placeholders;

class Add_two_Ints_serv : public rclcpp::Node //create a class and inharet from the rclcpp
{
    public: //puplic
        Add_two_Ints_serv() : Node("Add_two_ints") 
        {
            //assign the v and the methods
            server_= this->create_service<example_interfaces::srv::AddTwoInts>("add_two_ints",
                                        std::bind(&Add_two_Ints_serv::callback_add_two_int,this,_1,_2));
            RCLCPP_INFO(this->get_logger(),"the CPP add two int in running...");

        }   
    private:
    //define the v of the class and the private methods
        rclcpp::Service<example_interfaces::srv::AddTwoInts>::SharedPtr server_;

        void callback_add_two_int(const example_interfaces::srv::AddTwoInts::Request::SharedPtr request,
                                  const example_interfaces::srv::AddTwoInts::Response::SharedPtr responce )
        {
            responce->sum= request->a + request->b ;
            RCLCPP_INFO(this->get_logger(),"%d + %d = %d" , (int) request->a , (int) request->b , (int) responce->sum );
        }
};

int main(int argc,char **argv)
{
    rclcpp::init(argc,argv);
    auto node = std::make_shared<Add_two_Ints_serv>(); //create the shared pointer of the my node
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;

}

File: src/my_py_pkg/my_py_pkg/add_two_ints_serv.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts

class Add_Two_Ints_Node(Node): #create the class and inharet from node
    def __init__(self): 
        super().__init__("add_two_ints_serv") #call the init from the Node class 
        self.server_= self.create_service(AddTwoInts,"add_two_ints",self.callback_addtwoints)
        self.get_logger().info("add_two_int_service run....")



    def callback_addtwoints(self,request: AddTwoInts.Request, responce: AddTwoInts.Response):
        responce.sum = request.a + request.b
        self.get_logger().info(f"{request.a}+{request.b}={responce.sum}")
        return responce



def main(args=None):
    rclpy.init(args=args)   #init the communication
    node=Add_Two_Ints_Node()   #create a object from the class
    rclpy.spin(node)        #make the node spin
    rclpy.shutdown()        #shutdow the node



#if the name main called in the terminal call the main function
if __name__ == "__main__":
    main()